lib/services/database/comment_repository_service.dart

```
@@ -0,0 +1,124 @@
+  /// post with id [parentPostId].
+  /// It will also update the number of comments of the post.
+  /// This is done atomically.
+  ///
+  /// The method returns the id of the comment that was added.
+  Future<CommentIdFirestore> addComment(
+    PostIdFirestore parentPostId,
+    CommentData commentData,
+  ) async {
+    // Generate a new reference for the comment
+    // Although generated locally, the new id can be considered unique
+    // https://stackoverflow.com/questions/54268257/what-are-the-chances-for-firestore-to-generate-two-identical-random-keys
+    final newCommentRef = _commentsSubCollection(parentPostId).doc();
+
+    // Create a batch write to perform the operations atomically
+    final batch = _firestore.batch();
+
+    batch.set(newCommentRef, commentData.toDbData());
+
+    final postDocRef = _postDocument(parentPostId);
+    batch.update(
+      postDocRef,
+      {PostData.commentCountField: FieldValue.increment(1)},
+    );
+
+    await batch.commit();
+
+    return CommentIdFirestore(value: newCommentRef.id);
+  }
+
+  /// This method will delete the comment with id [commentId] from the
+  /// post with id [parentPostId].
+  /// It will also update the number of comments of the post.
+  /// This is done atomically.
+  /// If the comment does not exist, the method will do nothing.
+  Future<void> deleteComment(
+    PostIdFirestore parentPostId,
+    CommentIdFirestore commentId,
+  ) async {
+    await _firestore.runTransaction((transaction) async {
+      await _deleteComment(parentPostId, commentId, transaction);
+    });
+  }
+
+  // Concrete implementation of the deletion of a comment
+  Future<void> _deleteComment(
+    PostIdFirestore parentPostId,
+    CommentIdFirestore commentId,
+    Transaction transaction,
+  ) async {
+    final commentRef =
+        _commentsSubCollection(parentPostId).doc(commentId.value);
+
+    final commentDoc = await transaction.get(commentRef);
+
+    if (!commentDoc.exists) {
+      return;
```

We probably want to error in that case right ? I don't really like the idea of silently succeeding the operation of removing a comment that does not exist.
Is there a particular reason behind that choice ?

---

At first I though it was the best option to simply return.
But thinking again about it I agree with you.
Fix in d0a6592de9d506826ca7439a4206255dfbcabf79.